android/src/main/kotlin/com/onegini/mobile/sdk/flutter/useCases/ResourceRequestUseCase.kt

```diff
@@ -25,10 +27,38 @@ import javax.inject.Singleton
 @Singleton
 class ResourceRequestUseCase @Inject constructor(private val oneginiSDK: OneginiSDK) {
   operator fun invoke(type: ResourceRequestType, details: OWRequestDetails, callback: (Result<OWRequestResponse>) -> Unit) {
-    val resourceClient = getOkHttpClient(type)
-    val request = buildRequest(details)
+    val pathResult = getCompleteResourceUrl(details.path)
 
-    performCall(resourceClient, request, callback)
+    // Additional check for valid url
+    when {
+      pathResult.isFailure -> pathResult.exceptionOrNull()?.let { callback(Result.failure(it)) }
+      pathResult.isSuccess -> {
+        pathResult.getOrNull()?.let {
+          val resourceClient = getOkHttpClient(type)
+          val request = buildRequest(details, it)
+
+          performCall(resourceClient, request, callback)
+        }
+      }
+    }
+  }
+
+  private fun getCompleteResourceUrl(path: String): Result<String> {
+    val resourceBaseUrl = oneginiSDK.oneginiClient.configModel.resourceBaseUrl
+
+    return when {
+      isValidUrl(path) -> Result.success(path)
+      isValidUrl(resourceBaseUrl + path) -> Result.success(resourceBaseUrl + path)
```

Comment:
When the baseurl ends with a `/` and the path begins with a `/` then the url will have `//` this will likely get handled by the server, but we shouldn't be sending requests like that. Can we handle that somehow?
Kotlin has a URLBuilder class which could maybe be used here.